torchci/components/WorkflowDispatcher.tsx

```diff
@@ -141,3 +144,72 @@ export default function WorkflowDispatcher({
     </div>
   );
 }
+
+export function SingleWorkflowDispatcher({
+  sha,
+  jobName,
+}: {
+  sha: string;
+  jobName: string;
+}) {
+  let [runMoreJobsClicked, setRunMoreJobsClicked] = useState(false);
+
+  let session: any = useSession();
+  const router = useRouter();
+
+  let { repoOwner, repoName } = router.query;
+
+  if (!repoOwner) {
+    repoOwner = "pytorch";
+    repoName = "pytorch";
+  }
+
+  // extract workflow key from the jobName
+  const workflow = jobName.split(new RegExp("[/-]"))[0]?.toLowerCase()?.trim();
+
+  const repo = `${repoOwner}/${repoName}`;
+  if (
+    !session ||
+    !session.data ||
+    !session.data["accessToken"] ||
+    !(repo in SUPPORTED_WORKFLOWS) ||
+    !workflow ||
+    !(workflow in SUPPORTED_WORKFLOWS[repo])
+  ) {
+    return <></>;
+  }
+
+  const accessToken = session.data["accessToken"];
+
+  // avoid commit data fetching if the user hasn't clicked the button
+  if (!runMoreJobsClicked) {
+    return (
+      <div>
+        <button onClick={() => setRunMoreJobsClicked(true)}>
+          Run more jobs?
+        </button>
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <WithCommitData
```

Comment:
I think we could refactor `CommitInfo` in:

* https://github.com/pytorch/test-infra/blob/main/torchci/pages/%5BrepoOwner%5D/%5BrepoName%5D/commit/%5Bsha%5D.tsx#L7
* https://github.com/pytorch/test-infra/blob/main/torchci/pages/%5BrepoOwner%5D/%5BrepoName%5D/pull/%5BprNumber%5D.tsx#L10

to use ``WithCommitData`.  These functions are also duplicated and can be refactored into a shared `CommitInfo` component.

It works for me locally as follows:

```
import { WithCommitData } from "components/WithCommitData";

export function CommitInfo({
  repoOwner,
  repoName,
  sha,
}: {
  repoOwner: string;
  repoName: string;
  sha: string;
}) {
  const { data: unstableIssuesData } = useSWR(`/api/issue/unstable`, fetcher, {
    dedupingInterval: 300 * 1000,
    refreshInterval: 300 * 1000, // refresh every 5 minutes
  });

  return (
    <div>
      <WithCommitData sha={sha} repoOwner={repoOwner} repoName={repoName}>
        {({ commit, jobs }) => (
          <div>
            <h2>{commit.commitTitle}</h2>
            <CommitStatus
              repoOwner={repoOwner}
              repoName={repoName}
              commit={commit}
              jobs={jobs}
              isCommitPage={true}
              unstableIssues={
                unstableIssuesData ? unstableIssuesData.issues : []
              }
            />
          </div>
        )}
      </WithCommitData>
    </div>
  );
}
```

This could be done in a separate PR though because the current change works as it is.